java-checks/src/main/java/org/sonar/java/checks/ConstructorCallingOverridableCheck.java

```diff
@@ -87,7 +86,7 @@ public void visitMethodInvocation(MethodInvocationTree tree) {
       }
       if (isInvocationOnSelf) {
         Symbol symbol = methodIdentifier.symbol();
-        if (isOverridableMethod(symbol) && isMethodDefinedOnConstructedType(symbol)) {
+        if (isOverridableMethod(symbol) && isMethodDefinedOnConstructedType(symbol) &&!isInsideAnonymousClass(methodIdentifier)) {
```

Comment:
There is a simpler approach than what you implemented. It works perfectly, but it does not take advantage of the `ConstuctorBodyVisitor` being a `BaseTreeVisitor`.

I would remove the condition you added here, which require exploring parents in order to determine if you are part of a lambda or a new class tree, and replace it by simply overriding the tree visitors of the lambda and newClassTree visitors, skipping their analysis:

```
   @Override
    public void visitLambdaExpression(LambdaExpressionTree lambdaExpressionTree) {
      // skip lambdas
    }

    @Override
    public void visitNewClass(NewClassTree tree) {
      // skip new class
    }
```

Once done, no need to add extra logic, the exploration of these trees will be skipped, avoiding raising FPs in the process. The rest of your code can also be removed.